File: csharp/src/FuelPricePipeline.Domain/DieselFuelPrice.cs
using System.ComponentModel.DataAnnotations;

namespace FuelPricePipeline.Domain
{
    public class DieselFuelPrice
    {
        [Required, StringLength(10)]
        public string ProductCode { get; set; } = string.Empty;

        [Required]
        public string ProductName { get; set; } = string.Empty;

        [Required, StringLength(10)]
        public string AreaCode { get; set; } = string.Empty;

        [Required]
        public string AreaName { get; set; } = string.Empty;

        [Required]
        public DateTime Period { get; set; }

        [Range(0, 1000)]
        public decimal Value { get; set; }

        [Required]
        public string Unit { get; set; } = string.Empty;
    }
}

File: go/domain/diesel_fuel_price.go
package domain

import (
	"time"
)

type DieselFuelPrice struct {
	ID          int       `json:"id"`
	ProductCode string    `json:"product_code"`
	AreaCode    string    `json:"area_code"`
	Period      time.Time `json:"period"`
	Value       float64   `json:"value"`
	Unit        string    `json:"unit"`
	ProductName string    `json:"product_name"`
	AreaName    string    `json:"area_name"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}
